File: 프로그래머스/0/181913. 문자열 여러 번 뒤집기/문자열 여러 번 뒤집기.java
class Solution {
    public String solution(String my_string, int[][] queries) {
        String answer = "";
        char[] chars = my_string.toCharArray();
        for (int i=0; i<queries.length; i++) {
            for (int j=0; j<(queries[i][1] - queries[i][0] + 1) / 2; j++) {
                char temp = chars[queries[i][0]+j];
                chars[queries[i][0]+j] = chars[queries[i][1]-j];
                chars[queries[i][1]-j] = temp;
            }
        }
        answer = String.valueOf(chars);
        return answer;
    }
}

File: 프로그래머스/0/181913. 문자열 여러 번 뒤집기/문자열 여러 번 뒤집기.py
def solution(my_string, queries):
    answer = ''
    chars = list(my_string)
    for query in queries:
        start_index, end_index = query[0], query[1]
        for j in range((end_index - start_index + 1) // 2):
            temp = chars[start_index + j]
            chars[start_index + j] = chars[end_index - j]
            chars[end_index - j] = temp
    answer = ''.join(chars)
    return answer